### DISCUSSION TITLE
Get auth users data from relational table (React.js)

### DISCUSSION THREAD

**Discussion by akhil-dignizant** [supabase/supabase#27867] [Questions]
I have a table called messages, and this table has a column named user_id. This column contains the value of the authenticated users' IDs, which are linked to the primary key id in the authenticated users table. Therefore, the user_id column in my messages table is linked with the id column in the authenticated users table. When I fetch all data from the messages table, I need to populate the user_id field to get the corresponding user data. How can I do that?

**Comment by GaryAustin1** [comment]
You really need to go with the approach of the user management guides and have a public profile/user table populated by trigger functions on auth.users.  The auth schema cannot be accessed, joined or even used as an fk reference between multiple tables in the REST APIs.
https://supabase.com/docs/guides/auth/managing-user-data?queryGroups=language&language=js
